src/libraries/Microsoft.PowerFx.Core/Public/Values/BlobValue.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Diagnostics.Contracts;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+
+namespace Microsoft.PowerFx.Types
+{
+    public class BlobValue : FileValue
+    {
+        public BlobValue(IResourceManager resourceManager, int id)
+            : base(resourceManager, id)
+        {
+            Contract.Assert(IRContext.ResultType == FormulaType.Blob);
+        }
+
+        internal BlobValue(IRContext irContext)
+            : base(null, -2)
```

Comment:
>-2 [](http://example.com/codeflow?start=25&length=2)

Curious: what is this value?

Reply:
Some invalid value
I could define is as a constant to be explicit on the intent

Reply:
If this constructor will always fail, why do we need it? Shouldn't we always have a resource manager?

Reply:
Correct, no need :)